### ISSUE TITLE
Adding group by functionality in table structure itself.

### ISSUE THREAD

**Issue by Achilles-96** [phpmyadmin/phpmyadmin#12031]
We know that group by is very frequently used function. So, we can add a small `G` icon beside column name in table itself. On clicking it, a new table is displayed grouped by this column and all the eligible aggregate functions in each column.

**Comment by madhuracj** [comment]
So do you suggest showing columns for combination of all aggregate functions for all eligible columns? That will be quite a lot of columns.
Btw, though not exactly the same, one click aggregator can be found at the table structure page for each column. Look for "Distinct values" link (likely hidden under "More" in "Action" column)

**Comment by Achilles-96** [comment]
Yes, there will be too many columns if we display all the aggregate functions in one go. But we will display only count(*) initially and rest of columns(which are aggregate functions of some other column) will be based on columns in which user is interested in. We can provide some interface which enables user to do this(may be a drop down above new table).

**Comment by Achilles-96** [comment]
@madhuracj , Any views?

**Comment by madhuracj** [comment]
So the initial one would be similar to `Distinct values` and the addition would be the interface to add more columns to the aggregation and select the aggregation function? Well, it would be interesting to see how the UI would be presented and accessed from. Especially, whether it can be done without cluttering the `Browse` view.

**Comment by Achilles-96** [comment]
It is not exactly similar to `Distinct values` as it considers only primary key. The following UI is what I am thinking - Once group by the icon (`G`) is clicked, there will be two dropdowns and a `go` button below `+options` button. The user selects a column in the first dropdown and the second one will be populated with valid aggregate functions. The user selects one from that as well and clicks `go` and a new column will be added. I know it sounds a bit like spreadsheets but I feel it will be quite useful. 

PS: Sorry for long delay.

**Comment by Achilles-96** [comment]
@madhuracj , Any views on this?
